cmd/kv-tcp/main.go

```diff
@@ -81,22 +86,46 @@ func main() {
 	store = serializable.NewTwoPhaseLockStore(store)
 	transactor := transactors.New(store, writer)
 
-	server{store, transactor}.serve(ln.(*net.TCPListener))
+	s := server{ln.(*net.TCPListener), store, transactor, make(chan bool, 1)}
```

Comment:
We should probably move to naming the fields here instead of using their order within the struct. I think it was fine when there were only two fields, but four seems like a bit much.